Playmate/View Controllers/Create Tab/SelectMapViewController.m

```diff
@@ -68,35 +96,25 @@ - (void)handleAlert:(NSError * _Nullable)error withTitle:(NSString *)title andOk
 
 #pragma mark - Search bar and geocode
 
-//- (void)searchBar:(UISearchBar *)searchBar textDidChange:(NSString *)searchText {
-//
-//}
-
 - (void)searchBarSearchButtonClicked:(UISearchBar *)searchBar
 {
+    [self geocodeLocationWithSearch:searchBar.text];
+    [searchBar resignFirstResponder];
+}
+
+- (void)geocodeLocationWithSearch:(NSString *)searchInput {
     APIManager *manager = [APIManager new];
-    [manager getGeocodedLocation:searchBar.text withCompletion:^(Location *loc, NSError *error) {
+    [manager getGeocodedLocation:searchInput withCompletion:^(Location *loc, NSError *error) {
         
         if (error == nil) {
             
             if (loc == nil) {
                 [self handleAlert:nil withTitle:@"Address not found." andOk:@"Ok"];
             }
             else {
-                // extract information from Location object
-                // recenter map and put pin
-                
                 CLLocationCoordinate2D centerCoord = CLLocationCoordinate2DMake([loc.lat doubleValue], [loc.lng doubleValue]);
                 
-                MKCoordinateRegion region = MKCoordinateRegionMake(centerCoord, MKCoordinateSpanMake(0.1, 0.1));
-                [self.mapView setRegion:region animated:false];
-                
-                MKPointAnnotation *annotation = [[MKPointAnnotation alloc] init];
-                [annotation setCoordinate:centerCoord];
-                [annotation setTitle:loc.locationName];
-                [self.mapView addAnnotation:annotation];
-                
-                // send location back to filters or create view controller
+                [self dropPinOnMapAt:centerCoord withName:loc.locationName];
```

Comment:
I like how you create functions to avoid huge complex logic inside the closure. 

Should we worry about  calling `self` inside a closure? due to memory retain cycle? 
Did you got any warning in Xcode?